runtime/src/main/java/org/corfudb/runtime/object/CorfuCompileProxy.java

```diff
@@ -161,6 +162,20 @@ public CorfuCompileProxy(CorfuRuntime rt, UUID streamID, Class<T> type, Object[]
                 return TransactionalContext.getCurrentContext()
                         .access(this, accessMethod, conflictObject);
             } catch (Exception e) {
+                if (e instanceof TransactionAbortedException) {
```

Comment:
It -is- different. In a regular access, you may want to keep retrying, because you get a new stream-tail from the sequencer each time. 

in a TX context, on the other hand, we only want to try once against each trim-offset. If that fails, it means that the TX is trying to get to a snapshot which has been trimmed, and will just keep failing.